webui/ocw/lib/db.py

```diff
@@ -54,47 +55,53 @@ def __update_run():
     from . import EC2db
     from . import azure
     from . import gce
+    global update_error
 
     '''
     Each update is using Instance.active to mark the model is still availalbe on CSP.
     Instance.state is used to reflect the "local" state, e.g. if someone triggered a delete, the
     state will moved to DELETING. If the instance is gone from CSP, the state will set to DELETED.
     '''
+    try:
+        instances = Azure().list_resource_groups()
+        logger.info('Got {} resources groups from Azure'.format(len(instances)))
+        azure.sync_instances_db(instances)
```

Comment:
maybe better to add this try/catch thingy into each provider ? this patch for sure an improvement into comparison  what we had before ( we getting at least some error handling ) but still - you don't want to fail Azure DB update when you having troubles with Amazon , right ?

Reply:
hm, but doing it in each provider, would need a additional mechanize to announce the errors back to this update block.
I clould do a single try/catch per provider.

Reply:
Thinking twice on it. I actually need to investigate in any case, if there is an exception on update().
Event if it isn't that nice, but I could stop continue.
We could think on sending an email.

Reply:
as I said this version is already improvement , so it's actually fine. But imagine you have issue with update Azure and Amazon simultaneously :) with current code you will be not even aware about issue with Amazon update before you will not fix Azure issue :)  again I am ready to merge as it is because it would be improvement anyway .
Sending emails would be cool .  Do you want to do it in this PR or separate one ?

Reply:
Agree, lets do it in a separate one. I think I would then go with a  single thread per provider. So we also have a time improvement on updating.